src/Agent.Listener/Configuration/ConfigurationProvider.cs

```diff
@@ -308,6 +314,34 @@ private async Task<List<DeploymentMachine>> GetDeploymentTargetsAsync(AgentSetti
 
             return machines;
         }
+
+        private string GetAzureSubscriptionId()
```

Comment:
the caller `AddAgentAsync` is an async task.
change this to be an async task as well.
```C#
private async Task<string> GetAzureSubscriptionIdAsync()
{
     azureSubscriptionId  = await httpClient.GetStringAsync(imdsUri);
}
```

Reply:
done